libraries/chain/include/eosio/chain/chain_config.hpp

```diff
@@ -419,6 +512,34 @@ inline DataStream &operator>>(DataStream &s, eosio::chain::data_entry<eosio::cha
    return s;
 }
 
+/**
+ * @brief This is for unpacking data_entry<chain_config_v2, ...>
+ * that is used as part of unpacking data_range<chain_config_v2, ...>
+ * @param s datastream
+ * @param entry contains config reference and particular id
+ * @throws unsupported_feature if protocol feature for particular id is not activated
+ */
+template <typename DataStream>
+inline DataStream &operator>>(DataStream &s, eosio::chain::data_entry<eosio::chain::chain_config_v2, eosio::chain::config_entry_validator> &entry){
+   using namespace eosio::chain;
+
+   EOS_ASSERT(entry.is_allowed(), unsupported_feature, "config id ${id} is not allowed", ("id", entry.id));
+
+   switch (entry.id){
+      case chain_config_v2::max_sync_call_depth_id:
+      fc::raw::unpack(s, entry.config.max_sync_call_depth);
+      break;
+      case chain_config_v2::max_sync_call_data_size_id:
+      fc::raw::unpack(s, entry.config.max_sync_call_data_size);
+      break;
+      default:
+      eosio::chain::data_entry<chain_config_v1, config_entry_validator> base_entry(entry);
+      fc::raw::unpack(s, base_entry);
+   }
+
+   return s;
+}
+
```

Comment:
Is this a duplicate? Looks the same as the one above.

Reply:
It's not a duplicate. It is for v2. The one above is for v1.

Reply:
I mean lines 385-412 above

Reply:
lines 385 - 412 are for `operator<<` (packing), here are for `operator>>` (unpacking).